api/app/signals/apps/api/generics/serializers.py

```diff
@@ -0,0 +1,39 @@
+from rest_framework import serializers
+from rest_framework.exceptions import PermissionDenied
+from rest_framework.fields import empty
+
+
+class SIAModelSerializer(serializers.ModelSerializer):
```

Comment:
I would like to go over the reasoning behind the implementation of permission checks in the serializer layer as opposed to the view layer.

Having said that, this looks like a nice general solution to linking permissions to actions over the API.